aips/aip-hybrid-nft-standard.md

```diff
@@ -0,0 +1,352 @@
+    deposit_function: Option<FunctionInfo>
+): ConstructorRef
+```
+The `create` function establishes a new hybrid collection with specified parameters for both NFT and fungible asset components. It creates an object that will control the collection for minting and burning operations. The function allows customization of collection metadata, NFT properties, and fungible asset characteristics.
+
+2. **Treasury Management**
+```move
+public entry fun mint_to_treasury(
+    caller: &signer,
+    collection: Object<HybridCollection>,
+    amount: u64
+)
+```
+Treasury management functions handle the minting and distribution of assets. The `mint_to_treasury` function is particularly efficient as it bypasses NFT minting, making it suitable for initial supply creation. This is complemented by functions for treasury withdrawals and user distributions:
+
+```move
+public fun remove_from_treasury(
+    caller: &signer,
+    collection: Object<HybridCollection>,
+    amount: u64
+): FungibleAsset
+
+public entry fun send_from_treasury_to_user(
+    caller: &signer,
+    collection: Object<HybridCollection>,
+    receiver: address,
+    amount: u64
+)
+```
+
+### Asset Transfer Operations
+
+Transfer operations handle the movement of both fungible and non-fungible assets while maintaining the required ratio between them. These functions implement the core hybrid asset mechanics.
+
+1. **Standard Transfer**
+```move
+public entry fun transfer<T: key>(
+    caller: &signer,
+    token: Object<T>,
+    receiver: address
+)
+```
+The standard transfer function handles NFT transfers while automatically managing the associated fungible assets. This function ensures atomic execution of both NFT and fungible token transfers.
+
+2. **Fungible Asset Operations**
+```move
+public fun withdraw<T: key>(
+    store: Object<T>,
+    amount: u64,
+    transfer_ref: &fungible_asset::TransferRef
+): FungibleAsset
+
+public fun deposit<T: key>(
+    store: Object<T>,
+    fa: FungibleAsset,
+    transfer_ref: &fungible_asset::TransferRef
+)
+```
+These functions handle fungible asset transfers while managing the associated NFT minting and burning operations. When fungible assets are transferred:
+- If the recipient accumulates enough tokens for a "full NFT," an NFT is minted to their account
+- If the sender's balance falls below the threshold for a "full NFT," their NFT is burned
+
+3. **Special Transfer Operations**
+```move
+public fun deposit_with_ref<T: key>(
+    reveal_ref: &RevealRef,
+    store: Object<T>,
+    fa: FungibleAsset,
+    transfer_ref: &fungible_asset::TransferRef
+): vector<ConstructorRef>
+```
+Special transfer operations provide additional control over the minting and burning behavior, useful for specific scenarios such as DEX interactions or marketplace trades.
+
+### NFT Management
+
+NFT management functions control the revelation and modification of NFT metadata and properties.
+
+1. **Reveal Operations**
```

Comment:
This seems optional in this standard. Should this just be an extension instead of the core standard?